awscli/customizations/s3/utils.py

```diff
@@ -217,10 +218,23 @@ def get_file_stat(path):
     """
     try:
         stats = os.stat(path)
-        update_time = datetime.fromtimestamp(stats.st_mtime, tzlocal())
-    except (ValueError, IOError) as e:
+    except IOError as e:
         raise ValueError('Could not retrieve file stat of "%s": %s' % (
             path, e))
+
+    try:
+        update_time = datetime.fromtimestamp(stats.st_mtime, tzlocal())
+    except ValueError as e:
+        # Python's fromtimestamp raises value errors when the timestamp is out
+        # of range of the platform's C localtime() function. This can cause
+        # issues when syncing from systems with a wide range of valid timestamps
+        # to systems with a lower range. Some systems support 64-bit timestamps,
+        # for instance, while others only support 32-bit. We set the time to be
+        # UNIX epoch time so that file size will take precedence in sync.
+        LOG.warning("Attempting to access file with out of range timestamp. "
+                    "Passing epoch time as timestamp.")
+        update_time = datetime(1970, 1, 1)
```

Comment:
It will treat the file as being as old as possible, so it will never be synced unless it isn't already present or if the s3 lastmodified somehow gets set to lower. The warning and non-zero rc will alert the customer to the issue. Setting it to None would require a customization to sync strategies that rely on time, including any custom sync strategies that a customer might have.